### PR TITLE
[FIX] test_website_modules: fix configurator_flow

### PR DESCRIPTION
**Pull request by Xavier-Do**
For a strange reason, another pull request revealed some errors in this test/

- `#top_menu a[href^='${url}'` is not a valid selector
- There are multiple `button.btn-primary` on the page

### CODE REVIEW

**Comment by rdeodoo** [comment]
> There are multiple button.btn-primary on the page

Just to know, how is this true tho?
image

The testing engine should only look for non hidden elements, and only in modal if a modal is open, no (and we know the modal is open because of previous step)?

Otherwise, it's a failure in the testing engine itself no?

**Comment by Xavier-Do** [comment]
You are right, this part of the tests was not failing a deterministic way and I made an invalid assumption.
I tried to understand manually why it may be stuck there and found multiple button.btn-primary on the page. Since it was 
a recurrent error to randomly select the wrong element in the page, I assumed it was linked to that.

Trying to start the test locally I had another issue, the invalid selector, I still don't get why there is not such an issue on runbot but I got the error locally.

The real issue was that I made the reload much slower by making the ormcache key match the get_current_website. In this case, since the session is set, it will try to render /web with the new created website. The page is much slower and the step times out before the page load.

But I introduced the modification on the tour at the same time I introduced https://github.com/odoo-dev/odoo/commit/0de37d75443e190bd467ec101f42a14f3991e361 in my branch, and it was what fixed the tour in the next build. I falsely assumed that my modification on the tour worked.

File: addons/test_website_modules/static/tests/tours/configurator_flow.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ registry.category("web_tour.tours").add('configurator_flow', {
                 }
             }
             for (const url of ['/', '/event', '/slides', '/pricing', '/blog/', '/blog/', '/contactus']) {
-                if (!$iframe.contents().find(`#top_menu a[href^='${url}'`).length) {
+                if (!$iframe.contents().find(`#top_menu a[href^='${url}']`).length) {
```
Comment (rdeodoo) [inline addons/test_website_modules/static/tests/tours/configurator_flow.js:91]: @detrouxdev I am not sure who wrote this test as it's a multi author commit including you but maybe you know: I have the impression that this `run()` is actually never tested, is it?
From what I understand, won't the testing engine just find the `trigger` here (which succeed) and then try to find the next trigger (and at the same time start the run code) but since there is none, it will just mark the test as succeed, actually before the run is even executed?
Generally, when we check something in a custom `run()`, at the end we add a class or something so we can check its presence in the next step (to be sure we went through the full run and actually indirectly wait for it to be executed).

A bit like the `assertCssVariable` which IMO seems to be wrong as it does a check in the run but the next step is not waiting for it, so it will crash correctly because there is many steps after it taking some time which will leave time for the JS to actually crash, by chance?

tl;dr: The content of a `run()` method is not awaited to proceed to the next step. The next step's trigger of a tour should thus never be found before running the current step, which is the case here: nothing is ensuring that this step operation is done properly.

I wrote a wip back in the days to ensure the next step is not reachable but it seems to be hard to make automatic
https://github.com/odoo-dev/odoo/commit/2ef6ca12894ff46fe6c32c13fe39f866c95a70fd

Comment (rdeodoo) [reply addons/test_website_modules/static/tests/tours/configurator_flow.js:91]: Note that
```js
$iframe.contents().find(`#top_menu a[href^='${url}']`
```
Is not crashing from what we tested with ARD, jquery is magical enough to fix the selector for you somehow and find the element.

We will check to be sure what I said is right: run methods are not awaited and next step will ignore it / not wait before next step

image

Comment (Xavier-Do) [reply addons/test_website_modules/static/tests/tours/configurator_flow.js:91]: I actually understand what's going on, i'm on chrome 113 locally 

This is actually also breaking on runbot in the latest chrome build 
https://runbot.odoo.com/runbot/build/36161837
